state/src/traits.rs

```diff
@@ -177,4 +177,5 @@ pub trait TopState {
 pub trait StateWithCache {
     /// Commits our cached account changes into the trie.
     fn commit(&mut self) -> StateResult<H256>;
+    fn commit_and_unwrap(self) -> StateResult<(StateDB, H256)>;
```

Comment:
Let's think about another name.
`unwrap` doesn't give a feeling that it moves `State` to `StateDB`